pkg/controller/tas/resource_flavor.go

```diff
@@ -109,6 +111,12 @@ func (h *nodeHandler) Update(ctx context.Context, e event.UpdateEvent, q workque
 	if !isOldNode || !isNewNode {
 		return
 	}
+
+	if !checkNodeSchedulingPropertiesChanged(oldNode, newNode) {
+		ctrl.LoggerFrom(ctx).V(5).Info("Skipping node update as it doesn't affect TAS scheduling", "node", newNode.Name)
+		return
+	}
```

Comment:
Could you replace this check with a more efficient way? (w/o DeepCopy)

```go
var nodeSemantic = conversion.EqualitiesOrDir (nodeConditionEqual)

func nodeConditionEqual(a, b *corev1.NodeCondition) bool {
  return a.Type == b.Type && a.Status == b.Status && a.LastTransitionTime.Time.Equal(b.LastTransitionTime.Time) && a.Reason == b.Reason && a.Message == b.Message
}
```

```suggestion
	if !nodeSemantic.DeepEqual(oldNode, newNode) {
		ctrl.LoggerFrom(ctx).V(5).Info("Skipping node update as new Node is semantically same as old Node", "node", newNode.Name)
		return
	}
```

Reply:
Yes, but I was worried that we might forget to address something if a new field is added to a Node. However, it seems like people prefer to avoid DeepCopy, so I'll go with that.

Reply:
That's fair. In that case, we could use the following to mitigate risk and avoid high-computation deepcopy.
(The following DeepCopy will be executed only for NodeCondition).

```go
var nodeSemantic = conversion.EqualitiesOrDir (nodeConditionEqual)

func nodeConditionEqual(a, b corev1.NodeCondition) bool {
  aCopy, bCopy := a.DeepCopy(), b.DeepCopy()
  aCopy.LastHeartbeatTime, bCopy.LastHeartbeatTime = metav1.Time{}, metav1.Time{}
  return equality.Semantic.DeepEqual(aCopy, bCopy)
}
```